smack-extensions/src/main/java/org/jivesoftware/smackx/message_correct/MessageCorrectExtension.java

```diff
@@ -83,11 +81,13 @@ public String getElementName() {
      * @see org.jivesoftware.smack.packet.PacketExtension#toXML()
      */
     @Override
-    public CharSequence toXML() {
-        final StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append('<').append(getElementName()).append(' ').append(JID_TAG).append('=').append('\'')
-                .append(getJidInitialMessage()).append('\'').append(" xmlns='").append(getNamespace()).append("'/>");
-        return stringBuilder.toString();
+    public XmlStringBuilder toXML() {
+        XmlStringBuilder xml = new XmlStringBuilder(this, getNamespace());
```

Comment:
Just `new XmlStringBuilder(this)` will do the right thing, i.e. adding the opening tag and the xmlns attribute.